### PR TITLE
Engine addpolicy api

### PR DESCRIPTION
**Pull request by juliancnn**
|Related issue|
|---|
|#18524|

## Description

This PR adds a new endpoind to manage the policies, these will stop being an asset in the catalog to become an internal construction, managed externally through the api, allowing to create policies and modify them by api.
A hash is also added to be able to know if the api that is currently running is the same as the one we have in memory.

### List policies
```
vagrant@engine:~$ wazuh-engine policy list
- policy/test1/0
- policy/wazuh/0
- policy/test/0
```

#### Create policy
```bash 
# Add new policy (with default integrations and parents)
vagrant@engine:~$ wazuh-engine policy add -p policy/test/0

# List policies
vagrant@engine:~$ wazuh-engine policy list
- policy/wazuh/0
- policy/test/0

# Get the new policy in yml
vagrant@engine:~$ wazuh-engine policy get -p policy/test/0 -n system user wazuh
policy: policy/test/0
hash: 11464515449720324140
assets:
  - integration/wazuh-core/0
  - integration/syslog/0
  - integration/system/0
  - integration/windows/0
  - integration/apache-http/0
  - integration/suricata/0
default_parents:
  - user: decoder/integrations/0
  - wazuh: decoder/integrations/0

```

### List assets of policy

```
# By default only returns the user assets
vagrant@engine:~$ wazuh-engine policy asset-list -p policy/wazuh/0

# If we want the specific namespace assets
vagrant@engine:~$ wazuh-engine policy asset-list -p policy/wazuh/0 -n system
- integration/wazuh-core/0

vagrant@engine:~$ wazuh-engine policy asset-list -p policy/wazuh/0 -n wazuh
- integration/syslog/0
- integration/system/0
- integration/windows/0
- integration/apache-http/0
- integration/suricata/0

```

### Router: Check if policy is updated
```
vagrant@engine:~$ wazuh-engine router get | jq
[
  {
    "name": "default",
    "filter": "filter/allow-all/0",
    "policy": "policy/wazuh/0",
    "priority": 255,
    "policy_sync": "UPDATED"
  }
]

```

### CODE REVIEW

**Review by NahuFigueroa97 — CHANGES_REQUESTED** [review]
Very good work! Just one thing, the integration tests fail because in the given steps API of the router the policy is being created using the catalog, this changes in the current PR. They should generate the .proto of the policy for python and create the policy with the policy API.

```
# First Scenario
@given('I am authenticated with the router API "{name}"')
def step_impl(context, name: str):
    get_catalog = catalog_pb2.ResourceGet_Request()
    get_catalog.name = "policy"
    get_catalog.format = "yaml"
    context.result = API_CATALOG.send_command("resource", "get", MessageToDict(get_catalog))
```

File: src/engine/source/cmds/include/cmds/policy.hpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#include <json/json.hpp>
+
+namespace cmd::policy
+{
+namespace details
+{
+constexpr auto ORIGIN_NAME = "engine_integrated_policy_api";
+} // namespace details
+
+void runAddPolicy(std::shared_ptr<apiclnt::Client> client, const std::string& policyName);
```
Comment (soynof) [inline src/engine/source/cmds/include/cmds/policy.hpp:18]: Please, add documentation.

**Review by soynof — CHANGES_REQUESTED** [review]

File: src/engine/source/router/src/policyManager.cpp [hunk 2]
```diff
@@ -206,4 +206,15 @@ base::OptError PolicyManager::unSubscribeTraces(const std::string& name, std::si
...
 
+std::optional<std::string>  PolicyManager::getPolicyHash(const std::string& name) {
+    std::shared_lock<std::shared_mutex> lock(m_mutex);
+    auto it = m_policies.find(name);
+    if (m_policies.end() == it)
+    {
+        return std::nullopt;
+    }
+
+    return it->second[0].hash();
```
[the remark this answers is not in the document]
Comment (juliancnn) [reply src/engine/source/router/src/policyManager.cpp:217]: The hash is a mandatory requirement to raise a policy, seen from the policy manager's point of view, we return nullopt if the policy is not hashed (line 214 or the hash inside the optional).

File: src/engine/source/router/src/router.cpp [hunk 3]
```diff
@@ -198,7 +198,8 @@ std::vector<Router::Entry> Router::getRouteTable()
             const auto& priority = std::get<0>(route.second);
             const auto& filterName = std::get<1>(route.second);
             const auto& policyName = m_priorityRoute.at(priority).front().getTarget();
-            table.emplace_back(name, priority, filterName, policyName);
+            const auto& hash = m_policyManager->getPolicyHash(policyName).value_or("");
```
[the remark this answers is not in the document]
Comment (juliancnn) [reply src/engine/source/router/src/router.cpp:201]: This validation must be done here and not inside the policy manager, the policy manager must return null to indicate that the policy does not exist. This is just a double check.

File: src/engine/source/api/include/api/test/handlers.hpp [hunk 4]
```diff
@@ -171,9 +176,11 @@ std::optional<base::Error> deleteAssetFromCatalog(const std::shared_ptr<catalog:
  *
  * @return std::optional<base::Error> If an error occurs, returns the error. Otherwise, returns std::nullopt.
  */
+// TODO: CHECK IF THIS METODO EXIST!
```
[the remark this answers is not in the document]
Comment (juliancnn) [reply src/engine/source/api/include/api/test/handlers.hpp:179]: yes

**Review by NahuFigueroa97 — APPROVED** [review]
LGTM!

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**